source/extensions/filters/common/local_ratelimit/local_ratelimit_impl.cc

```diff
@@ -13,16 +13,76 @@ namespace Filters {
 namespace Common {
 namespace LocalRateLimit {
 
+SINGLETON_MANAGER_REGISTRATION(local_ratelimit_share_provider_manager);
+
+class DefaultEvenShareMonitor : public ShareProviderManager::ShareMonitor {
+public:
+  uint32_t tokensPerFill(uint32_t origin_tokens_per_fill) const override {
+    return std::ceil(origin_tokens_per_fill * share_);
+  }
+  void onLocalClusterUpdate(const Upstream::Cluster& cluster) override {
+    const auto number = cluster.info()->endpointStats().membership_total_.value();
```

Comment:
This is only called in the main thread. Because the PrioritySet here is only the main thread PrioritySet.
And accessing the `membership_total_` is atomic operation.

Reply:
Let's add another assert main-thread here.